ground/src/main/java/com/google/android/ground/model/locationofinterest/Point.java

```diff
@@ -17,10 +17,14 @@
 package com.google.android.ground.model.locationofinterest;
 
 import com.google.auto.value.AutoValue;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.GeometryFactory;
 
 /** The location of a single point on the map. */
 @AutoValue
 public abstract class Point {
+  private final GeometryFactory geometryFactory = new GeometryFactory();
```

Comment:
Should we instead be using JTS `Point` instead? Or should we just create our own topology classes in the model as I'm doing in https://github.com/google/ground-platform/pull/928? Sorry to flip-flop, would like to think about it well now since we won't have time to revisit for awhile.

Note that going forward we may want to consider keeping conversion logic separate from data classes.

Reply:
> Should we instead be using JTS Point instead? 

Yes. I decided to omit that from this PR, but the next one will delete this pointless `Point` class and use JTS directly.

But, if the broader plan is to use our own model, I think we should just do that. There's some benefit to having the representation on both sides be equivalent.

Reply:
We spoke. Let's go with our own geometry impl, but please feel free to merge this PR first and retrofit.